test/dates/io.jl

```diff
@@ -66,11 +66,15 @@ b2 = "96/Feb/1"
 # Here we've specifed a text month name, but given a number
 b3 = "96/2/15"
 @test_throws ArgumentError Dates.DateTime(b3,f)
-let err = try DateTime("2012-02-20T09:09:3.43i9") catch err; err end
+try
+    Dates.tryparse_internal(DateTime, "2012-02-20T09:09:3.43i9", Dates.ISODateTimeFormat, true)
```

Comment:
add a `@test false` here if the exception should happen all the time, or if it doesn't throw then this would pass silently